shepherd

```diff
@@ -23,18 +23,22 @@ update_services() {
       image_with_digest="$(docker service inspect "$name" -f '{{.Spec.TaskTemplate.ContainerSpec.Image}}')"
       image=$(echo "$image_with_digest" | cut -d@ -f1)
       echo "Trying to update service $name with image $image"
-      docker service update "$name" $detach_option $registry_auth --image="$image" > /dev/null
+      docker pull $image && docker service update "$name" $detach_option $registry_auth --image="$image" > /dev/null
```

Comment:
Well, this means *all images* are pulled to the machine that shepherd is running on. Which means quite some bandwidth and storage overhead.

As `docker manifest` is now available, I am really curious to try this out instead: https://stackoverflow.com/a/56316948

Would you like to adjust your PR accordingly?

Reply:
Thanks for your quick response. 

I going to read your link and will try to adjust my PR.

In the meantime, I have a question (maybe naive, sorry) will not the images be pulled to the nodes anyway? I mean, the nodes need it to run, what am I missing?

Reply:
Done!

Reply:
> Will not the images be pulled to the nodes anyway? I mean, the nodes need it to run, what am I missing?

Each image will only be pulled to the node the corresponding service runs on. With your pull approach, all images end on the node which runs the shepherd service as well.

Reply:
Oh, I understand now, thanks!